src/System.IO.Packaging/src/System/IO/Packaging/Invariant.cs

```diff
@@ -0,0 +1,246 @@
+        ///                  partial trust.  However, not shutting down in
+        ///                  partial trust is even worse: by definition a false condition
+        ///                  means we've hit a bug in avalon code and we cannot safely
+        ///                  continue.
+        /// </SecurityNote>
+        // [SecurityCritical, SecurityTreatAsSafe] - Removed for performance, OK so long as this class remains internal
+        internal static void Assert(bool condition, string invariantMessage, string detailMessage)
+        {
+            if (!condition)
+            {
+                FailFast(invariantMessage, detailMessage);
+            }
+        }
+
+        #endregion Internal Methods
+
+        //------------------------------------------------------
+        //
+        //  Internal Properties
+        //
+        //------------------------------------------------------
+
+        #region Internal Properties
+
+        /// <summary>
+        /// Property specifying whether or not the user wants to enable expensive
+        /// verification diagnostics.  The Strict property is rarely used -- only
+        /// when performance profiling shows a real problem.
+        /// </summary>
+        ///
+        // Default value is false on FRE builds, true on 
+        internal static bool Strict
+        {
+            get { return s_strict; }
+
+            set { s_strict = value; }
+        }
+
+        #endregion Internal Properties
+
+        //------------------------------------------------------
+        //
+        //  Private Methods
+        //
+        //------------------------------------------------------
+
+        #region Private Methods
+
+        /// <summary>
+        ///     Shuts down the process immediately, with no chance for additional
+        ///     code to run.
+        /// 
+        ///     In debug we raise a Debug.Assert dialog before shutting down.
+        /// </summary>
+        /// <param name="message">
+        ///     Message to display before shutting down the application.
+        /// </param>
+        /// <param name="detailMessage">
+        ///     Additional message to display before shutting down the application.
+        /// </param>
+        /// <SecurityNote>
+        ///     Critical: This code will close the current process.
+        ///     TreatAsSafe: This code is safe to call.
+        ///         Note that if this code were made to be callable publicly,
+        ///         we would have a potential denial-of-service vulnerability.
+        /// </SecurityNote>
+        // [SecurityCritical, SecurityTreatAsSafe] todo ew
+        // [SecurityCritical]
+        private // DO NOT MAKE PUBLIC OR INTERNAL -- See security note
+            static void FailFast(string message, string detailMessage)
+        {
+            Debug.Assert(false, string.Format("Invariant failure: {0}\n{1}", message, detailMessage));
```

Comment:
Ideally we'd use Environment.NewLine for newlines, rather than hardcoding "\r\n" or "\n".